### ISSUE TITLE
SimpleFormIterator: Clear items call to action

### ISSUE THREAD

**Issue by soullivaneuh** [marmelab/react-admin#8287]
**Is your feature request related to a problem? Please describe.**

Sometimes, we need to get an array field to be emptied. 

Currently, the only way to achieve that is to remove each row of the array input, one by one because the inner form iterator does not have any clear action available.

**Describe the solution you'd like**
A "Clear items" call to action that will remove all the items at once.

**Describe alternatives you've considered**
We did a custom `ArrayInput` that implement this functionalities:

```tsx
import React, { FC } from 'react';
import {
  ArrayInput as BaseArrayInput,
  ArrayInputProps as BaseArrayInputProps,
  useTranslate,
  Button,
} from 'react-admin';
import {
  useFormContext,
} from 'react-hook-form';
import {
  CachedOutlined,
} from '@mui/icons-material';

interface ArrayInputProps extends BaseArrayInputProps {
  enableClearItem?: boolean;
}

export const ArrayInput:FC<ArrayInputProps> = ({
  enableClearItem,
  children,
  source,
  ...props
}) => {
  const { setValue, getValues } = useFormContext();
  const translate = useTranslate();
  const inputValues = getValues(source);
  return (
    <>
      <BaseArrayInput
        source={source}
        {...props}
      >
        {children}
      </BaseArrayInput>
      {enableClearItem && inputValues?.length > 0 && (
        <Button
          fullWidth
          label={translate('actions.arrayClear')}
          onClick={() => setValue(source, [], { shouldDirty: true })}
        >
          <CachedOutlined />
        </Button>
      )}
    </>
  );
};
```

It does the trick, however the rendering is not really clean when used with a `helperText` prop:

![image](https://user-images.githubusercontent.com/1698357/197154249-ec149173-2aad-4a73-b063-4cd783547c7e.png)

We should override directly the `SimpleFormIterator` instead. However, we have to copy/paste all the original logic of the vendor one, quite cumbersome.

**Additional context**

Note: As this action may be more dangereous as the other, we may make it not available by default with a `enableClear` prop.

**Comment by slax57** [comment]
Seems like a great idea to me.
Any PR submission, or upvotes, are welcome!

**Comment by Seojun-Park** [comment]
Related PR is created :-)

**Comment by fzaninotto** [comment]
Fixed by #8302
